test/xchar-test.cc

```diff
@@ -102,11 +102,62 @@ struct custom_char {
   template <typename T>
   constexpr custom_char(T val) : value(static_cast<int>(val)) {}
 
-  operator char() const {
+  constexpr operator char() const {
     return value <= 0xff ? static_cast<char>(value) : '\0';
   }
+  constexpr bool operator<(custom_char c) const { return value < c.value; }
 };
 
+#  ifdef _LIBCPP_VERSION
```

Comment:
I think we should specialize the trait unconditionally here. Since relying on the standard one is discouraged we don't care whether it works or not with custom code unit types.

Reply:
Done